proto/ibc/core/channel/v1/upgrade.proto

```diff
@@ -6,6 +6,25 @@ option go_package = "github.com/cosmos/ibc-go/v7/modules/core/04-channel/types";
 
 import "gogoproto/gogo.proto";
 import "ibc/core/client/v1/client.proto";
+import "ibc/core/channel/v1/channel.proto";
+
+// Upgrade is a verifiable type which contains the relevant information
+// for an attempted upgrade. It provides the proposed changes to the channel
+// end, the timeout for this upgrade attempt and the last packet sequence sent
+// to allow the counterparty to block sends after the upgrade has started.
+message Upgrade {
+  ModifiableUpgradeFields proposed_upgrade = 1 [(gogoproto.nullable) = false];
+  UpgradeTimeout          timeout          = 2 [(gogoproto.nullable) = false];
+  uint64                  last_packet_sent = 3;
```

Comment:
I know we haven't put too much thought into this yet but it may be difficult (or even impossible) for any actor to determine the last packet sequence sent on a well known/canonical channel if the upgrade is expected to be initiated via on-chain gov - due to the grace period for deposit/voting.